frontend/js/pages/BattleDetail.js

```diff
@@ -1,46 +1,54 @@
 import React, { useEffect } from 'react';
 import { connect } from 'react-redux';
-import { useParams } from 'react-router-dom';
+import { useParams, Link } from 'react-router-dom';
 
 import CardTeam from 'components/CardTeam';
 
 import { getBattle } from '../actions/getBattle';
 import { getCurrentUser } from '../actions/getUser';
 import { orderTeamsByCurrentUser } from '../utils/battle-detail';
+import { denormalizeBattleData } from '../utils/denormalize';
 import Urls from '../utils/urls';
 
 function BattleDetail(props) {
   const { id } = useParams();
-  const { battle } = props.battle;
-  const { user } = props.user;
-
+  const { battle, battles, user } = props;
   useEffect(() => {
     if (!user) {
       props.getCurrentUser();
     }
-    props.getBattle(id);
   }, []);
 
-  if (!battle) {
-    return '';
-  }
+  useEffect(() => {
+    if (!battles) {
+      props.getBattle(id);
+    }
+  }, []);
 
-  const { current, other } = orderTeamsByCurrentUser(battle, user);
+  if (!battle && !battles) {
+    return 'loading';
+  }
+  const denormalizedData = denormalizeBattleData(id, battles, battle);
```

Comment:
This name could be improved

Reply:
All right! I'll improve it thanks :dancing_women: